lib/saluki-core/src/pooling/helpers.rs

```diff
@@ -223,16 +223,18 @@ macro_rules! pooled_newtype {
             ) -> Self {
                 Self {
                     strategy_ref,
-                    data: ::std::mem::ManuallyDrop::new(data),
+                    data: ::std::option::Option::Some(data),
                 }
             }
         }
 
         impl Drop for $name {
             fn drop(&mut self) {
                 // SAFETY: We never use `self.data` again since we're already dropping `self`.
-                let data = unsafe { ::std::mem::ManuallyDrop::take(&mut self.data) };
-                self.strategy_ref.reclaim(data);
+
+                if let ::std::option::Option::Some(data) = self.data.take() {
+                    self.strategy_ref.reclaim(data);
+                }
```

Comment:
For posterity: we had to do this because otherwise, trying to consume `self.data` (like we now do in `BytesBuffer::freeze`) becomes impossible as the compiler recognizes that we're emptying out `self.data` at some point, and then the drop logic here runs, and also tries to take the value... which is clearly not valid.

This is just a simpler, and safe, way to support `freeze` semantics with the lowest overhead we can reasonably hope for.